core/irCheck.ml

```diff
@@ -422,7 +424,11 @@ let ensure_effect_present_in_row ctx allowed_effects required_effect_name requir
 let ensure_effect_rows_compatible ctx allowed_effects imposed_effects_row occurrence =
   ensure
     (eq_types occurrence ctx (`Record allowed_effects, `Record imposed_effects_row))
-    ("Incompatible effects; Allowed:\n" ^ (Types.string_of_row allowed_effects) ^ "\nactual effects:\n" ^  (Types.string_of_row imposed_effects_row))
+    (let allowed_str = Types.string_of_row allowed_effects in
+     let actual_str  = Types.string_of_row ~refresh_tyvar_names:true
```

Comment:
Ugh. This is bogus. Should be `~refresh_tyvar_names:false`. Will fix and merge into master